From 55455a9f8a07b66280ae5c95580c1390720f4362 Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Mon, 20 Nov 2023 15:26:28 +0100
Subject: [PATCH 0687/1141] dt-bindings: misc: Fix make dtbs_check for
 st,stm32-omm.yaml

make ARCH=arm64 dtbs_check DT_SCHEMA_FILES=st,stm32-omm.yaml returns
the following:

  DTC_CHK arch/arm64/boot/dts/st/stm32mp257f-ev1-revB.dtb
/local/home/nxp11987/projects/linux-stm32/MP2/arch/arm64/boot/dts/st/stm32mp257f-dk.dtb: ommanager@40500000: 'power-domains' does not match any of the regexes: '^memory-controller@[a-f0-9]+$', '^spi@[a-f0-9]+$', 'pinctrl-[0-9]+'
	from schema $id: http://devicetree.org/schemas/misc/st,stm32-omm.yaml#
/local/home/nxp11987/projects/linux-stm32/MP2/arch/arm64/boot/dts/st/stm32mp257f-ev1.dtb: ommanager@40500000: 'power-domains' does not match any of the regexes: '^memory-controller@[a-f0-9]+$', '^spi@[a-f0-9]+$', 'pinctrl-[0-9]+'
	from schema $id: http://devicetree.org/schemas/misc/st,stm32-omm.yaml#
/local/home/nxp11987/projects/linux-stm32/MP2/arch/arm64/boot/dts/st/stm32mp257f-ev1-revB.dtb: ommanager@40500000: 'power-domains' does not match any of the regexes: '^memory-controller@[a-f0-9]+$', '^spi@[a-f0-9]+$', 'pinctrl-[0-9]+'
	from schema $id: http://devicetree.org/schemas/misc/st,stm32-omm.yaml#
Looking for device trees into /local/home/nxp11987/projects/linux-stm32/MP2/arch/arm64/boot/dts/st/external-dt/linux
scripts/Makefile.build:44: /local/home/nxp11987/projects/linux-stm32/MP2/arch/arm64/boot/dts/st/external-dt/linux/Makefile: No such file or directory
make[1]: *** No rule to make target '/local/home/nxp11987/projects/linux-stm32/MP2/arch/arm64/boot/dts/st/external-dt/linux/Makefile'.  Stop.

Add power-domain in binding and update exmaples.

Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Change-Id: I8119cd2029db6375b08d3823e3612d2830be1efc
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/340949
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
---
 Documentation/devicetree/bindings/misc/st,stm32-omm.yaml | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/Documentation/devicetree/bindings/misc/st,stm32-omm.yaml
+++ b/Documentation/devicetree/bindings/misc/st,stm32-omm.yaml
@@ -102,6 +102,9 @@ properties:
         - 0x2: swapped mode
         - 0x3: mux OCTOSPI1 and OCTOSPI2 to port 2
 
+  power-domains:
+    maxItems: 1
+
 patternProperties:
   "^spi@[a-f0-9]+$":
     type: object
@@ -142,6 +145,7 @@ examples:
       st,omm-mux = <0x0>;
       st,omm-cssel-ovr = <0x0>;
       feature-domains = <&rifsc STM32MP25_RIFSC_OCTOSPIM_ID>;
+      power-domains = <&CLUSTER_PD>;
 
       ranges = <0 0 0x40430000 0x400>,
                <1 0 0x40440000 0x400>;
@@ -159,6 +163,7 @@ examples:
         clocks = <&rcc CK_KER_OSPI1>;
         resets = <&rcc OSPI1_R>, <&rcc OSPI1DLL_R>;
         feature-domains = <&rifsc STM32MP25_RIFSC_OCTOSPI1_ID>;
+        power-domains = <&CLUSTER_PD>;
       };
 
       memory-controller@40440000 {
@@ -174,5 +179,6 @@ examples:
         clocks = <&rcc CK_KER_OSPI2>;
         resets = <&rcc OSPI2_R>, <&rcc OSPI2DLL_R>;
         feature-domains = <&rifsc STM32MP25_RIFSC_OCTOSPI2_ID>;
+        power-domains = <&CLUSTER_PD>;
       };
     };
